### PR TITLE
bug 1381516 - Add analysis to see how many keyed histograms are affected and for how long

### PR DESCRIPTION
**Pull request by chutten**

### CODE REVIEW

File: bug1381516.kp/knowledge.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+The whole child-process client aggregation thing was introduced by [bug 1218576](https://bugzilla.mozilla.org/show_bug.cgi?id=1218576) back in September of 2016 for Firefox 52. So that's the earliest this could have started.
+
+
+```python
+pings = Dataset.from_source("telemetry") \
+    .where(docType='main') \
+    .where(appVersion=lambda x: x.startswith("52")) \
+    .where(appUpdateChannel="nightly") \
+    .records(sc, sample=0.1)
```
Comment (harterrt) [inline bug1381516.kp/knowledge.md]: I've heard the `sample` param doesn't behave like you would expect. Specifically, this isn't expected to be a uniform sample of pings. However, it looks like you've found positive evidence of the behavior in question, so this shouldn't be detrimental to your analysis. Is this interpretation correct?

File: bug1381516.kp/knowledge.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+     ((u'IPC_SYNC_RECEIVE_MS', u'???'), 4227),
+     ((u'IPC_READ_MAIN_THREAD_LATENCY_MS', u'PCompositorBridge::Msg_DidComposite'),
+      3523),
+     ((u'NOTIFY_OBSERVERS_LATENCY_MS', u'document-element-inserted'), 3498),
+     ((u'IPC_WRITE_MAIN_THREAD_LATENCY_MS',
+       u'PCompositorBridge::Msg_PTextureConstructor'),
+      2231),
+     ((u'IPC_MESSAGE_SIZE', u'PBrowser::Msg_AsyncMessage'), 2083),
+     ((u'IPC_READ_MAIN_THREAD_LATENCY_MS', u'PBrowser::Msg_AsyncMessage'), 2031)]
+
```
Comment (harterrt) [inline bug1381516.kp/knowledge.md]: I don't really understand the conclusion here. Is it, "Yes, we've seen this recently but with different probes?"

File: bug1381516.kp/knowledge.md (hunk 1 above)
Comment (fbertsch) [inline bug1381516.kp/knowledge.md]: Drive-by comment: we should really *really* document `sample` and `limit` behavior in DTMO.

Comment (harterrt) [inline bug1381516.kp/knowledge.md]: Seems like it belongs in the Dataset API reference and not DTMO

https://bugzilla.mozilla.org/show_bug.cgi?id=1382205

Comment (chutten) [inline bug1381516.kp/knowledge.md]: Yup. This isn't intended to be a count, more just evidence of existence.

File: bug1381516.kp/knowledge.md (hunk 2 above)
Comment (chutten) [inline bug1381516.kp/knowledge.md]: That's precisely it.  I can reword.

**Comment by chutten** [comment]
@harterrt RFAL
